src/java/main/org/apache/jute/compiler/CGenerator.java

```diff
@@ -61,70 +61,88 @@ void genCode() throws IOException {
+            c.write("* with the License.  You may obtain a copy of the License at\n");
+            c.write("*\n");
+            c.write("*     http://www.apache.org/licenses/LICENSE-2.0\n");
+            c.write("*\n");
+            c.write("* Unless required by applicable law or agreed to in writing, software\n");
+            c.write("* distributed under the License is distributed on an \"AS IS\" BASIS,\n");
+            c.write("* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n");
+            c.write("* See the License for the specific language governing permissions and\n");
+            c.write("* limitations under the License.\n");
+            c.write("*/\n");
+            c.write("\n");
 
-        h.write("#include \"recordio.h\"\n");
-        for (Iterator<JFile> i = mInclFiles.iterator(); i.hasNext();) {
-            JFile f = i.next();
-            h.write("#include \""+f.getName()+".h\"\n");
-        }
-        // required for compilation from C++
-        h.write("\n#ifdef __cplusplus\nextern \"C\" {\n#endif\n\n");
+            h.write("#ifndef __" + mName.toUpperCase().replace('.', '_') + "__\n");
+            h.write("#define __" + mName.toUpperCase().replace('.', '_') + "__\n");
 
-        c.write("#include <stdlib.h>\n"); // need it for calloc() & free()
-        c.write("#include \""+mName+".h\"\n\n");
+            h.write("#include \"recordio.h\"\n");
+            for (Iterator<JFile> i = mInclFiles.iterator(); i.hasNext(); ) {
+                JFile f = i.next();
+                h.write("#include \"" + f.getName() + ".h\"\n");
+            }
+            // required for compilation from C++
+            h.write("\n#ifdef __cplusplus\nextern \"C\" {\n#endif\n\n");
 
-        for (Iterator<JRecord> i = mRecList.iterator(); i.hasNext();) {
-            JRecord jr = i.next();
-            jr.genCCode(h, c);
-        }
+            c.write("#include <stdlib.h>\n"); // need it for calloc() & free()
+            c.write("#include \"" + mName + ".h\"\n\n");
 
-        h.write("\n#ifdef __cplusplus\n}\n#endif\n\n");
-        h.write("#endif //"+mName.toUpperCase().replace('.','_')+"__\n");
+            for (Iterator<JRecord> i = mRecList.iterator(); i.hasNext(); ) {
+                JRecord jr = i.next();
+                jr.genCCode(h, c);
+            }
 
-        h.close();
-        c.close();
+            h.write("\n#ifdef __cplusplus\n}\n#endif\n\n");
+            h.write("#endif //" + mName.toUpperCase().replace('.', '_') + "__\n");
+        } catch (IOException e) {
+            throw e;
+        } finally {
+            if (h != null) {
+                try {
+                    h.close();
+                } catch (IOException ex) {
+                    throw ex;
+                } finally {
+                    if (c != null) {
+                        c.close();
+                    }
+                }
+            }
+            if (c != null) {
```

Comment:
It's sad that we have to close c twice (one here, one in the finally block earlier.). Findbugs is not happy if either of them is missing. It could be that the flow sensitive engine of findbug need an improvement.